src/gfx/core.js

```diff
@@ -220,8 +220,17 @@ export function baseShape(name, defaults, draw, notchOffset=null) {
             const { x, y } = util.topLeftPos(this, offset);
 
             if (this.opacity) ctx.globalAlpha = this.opacity;
+            if (offset.opacity) {
```

Comment:
It should probably be something like `ctx.globalAlpha = this.opacity * offset.opacity` (guarding against `undefined`) so that opacity properly combines. I guess this tries to avoid the layering effect, but in some sense we probably want to show that because it reflects the AST (to an extent).

Reply:
wait you mean putting line 222 and the if statement below together? or just change the if?

Reply:
I mean instead of either if statement, you want something like

```
const selfOpacity = typeof this.opacity === "undefined" ? 1.0 : this.opacity;
ctx.globalAlpha = offset.opacity * selfOpacity;
```

That way the view's opacity doesn't get completely overridden by the parent opacity, but failing to set the view opacity won't result in an opaque object either, it'll use the parent opacity.

and since this is used a lot, we probably want a helper function like setStroke: https://github.com/lidavidm/reduct-redux/blob/9cf580a930a7c450c2737b8dbaf90201df19c94f/src/gfx/primitive.js#L128

Reply:
Done. I put the function in util.js since I wasn't sure where else but open to suggestions!